source/fem-dem/gls_sharp_navier_stokes.cc

```diff
@@ -2516,431 +2642,414 @@ GLSSharpNavierStokesSolver<dim>::sharp_edge()
+                              // If we are here, the DOF is on a boundary.
+                              particle_close_to_wall = true;
+                              cell_2                 = cell;
+                              // If a boundary condition is already applied to
+                              // this DOF we skip it otherwise we impose a value
+                              // base on the velocity of the particle.
+                              if (this->zero_constraints.is_constrained(
+                                    global_index_overwrite) ||
+                                  this->nonzero_constraints.is_constrained(
+                                    global_index_overwrite))
+                                {
+                                  continue;
+                                }
                             }
-                        }
 
-                      cell_2->get_dof_indices(local_dof_indices_2);
-                      ib_done[global_index_overwrite] =
-                        std::make_pair(true, cell_2);
-
-                      bool skip_stencil = false;
-
-                      // Check if the DOF intersect the IB
-                      bool dof_on_ib = false;
-
-                      // Check if this dof is a dummy dof or directly on IB and
-                      // Check if the point used to define the cell used for the
-                      // definition of the stencil ("cell_2") is on a face
-                      // between the cell that is cut ("cell") and the "cell_2".
-                      bool point_in_cell = cell->point_inside(
-                        interpolation_points
-                          [stencil.number_of_interpolation_support_points(
-                             order) -
-                           1]);
-
-                      bool         dof_is_dummy = false;
-                      bool         cell2_is_cut;
-                      unsigned int ib_particle_id_2;
-                      std::tie(cell2_is_cut, ib_particle_id_2, std::ignore) =
-                        cut_cells_map[cell_2];
-                      if (cell2_is_cut || point_in_cell)
-                        {
-                          dof_is_dummy = true;
-                        }
+                          cell_2->get_dof_indices(local_dof_indices_2);
+                          ib_done[global_index_overwrite] =
+                            std::make_pair(true, cell_2);
+
+                          bool skip_stencil = false;
+
+                          // Check if the DOF intersect the IB
+                          bool dof_on_ib = false;
+
+                          // Check if this dof is a dummy dof or directly on IB
+                          // and Check if the point used to define the cell used
+                          // for the definition of the stencil ("cell_2") is on
+                          // a face between the cell that is cut ("cell") and
+                          // the "cell_2".
```

Comment:
Instead of defining what is cell_2 and cell, why not instead call cell_2 "stencil_cell" and cell "cut_cell". This would be more verbose